src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -60,5 +101,19 @@ private static bool HandleEndObject(JsonSerializerOptions options, ref ReadStack
             ApplyObjectToEnumerable(value, options, ref state.Current);
             return false;
         }
+
+        internal static void ApplyValueToDictionary<TProperty>(ref TProperty value, JsonSerializerOptions options, ref ReadStackFrame frame)
+        {
+            Debug.Assert(frame.IsDictionary());
+            ((List<TProperty>)frame.TempDictValues).Add(value);
```

Comment:
We should try not to have a `TempDictValues` and just add to the newly instantiated dictionary. We should be able to treat dictionary like we treat a list - add values as they are found -- not cache until later (unlike arrays and other collections types that do require a temp list).